ros2bag/ros2bag/verb/record.py

```diff
@@ -232,14 +279,17 @@ def main(self, *, args):  # noqa: D102
             custom_data=custom_data
         )
         record_options = RecordOptions()
-        record_options.all = args.all
+        record_options.all_topics = args.all_topics or args.all
         record_options.is_discovery_disabled = args.no_discovery
         record_options.topics = args.topics
         record_options.rmw_serialization_format = args.serialization_format
         record_options.topic_polling_interval = datetime.timedelta(
             milliseconds=args.polling_interval)
         record_options.regex = args.regex
-        record_options.exclude = args.exclude
+        record_options.exclude_regex = args.exclude_regex
+        record_options.exclude_topics = args.exclude_topics if args.exclude_topics else []
+        record_options.exclude_services = \
+            convert_service_to_service_event_topic(args.exclude_services)
```

Comment:
One more finding. It seems there is a wrong name for the parameter after renaming
```suggestion
        record_options.exclude_service_events = \
            convert_service_to_service_event_topic(args.exclude_services)
```

Reply:
The definition as below

https://github.com/Barry-Xu-2018/rosbag2/blob/5c2006503d91ce47b29cf8b89782954fdf601d6b/rosbag2_py/src/rosbag2_py/_transport.cpp#L361

Not use the same variable name.    

Besides, I use comments to explain variables (services, exclude_service_events). Perhaps it would be better to make the variable names clearer (e.g. services -> service_event_topics, exclude_service_events -> exclude_service_event_topics). What do you think?
```
struct RecordOptions
{
public:
  bool all_topics = false;
  bool all_services = false;
  bool is_discovery_disabled = false;
  std::vector<std::string> topics;
  std::vector<std::string> services;  // service event topic
  std::vector<std::string> exclude_topics;
  std::vector<std::string> exclude_service_events;  // service event topic
  std::string rmw_serialization_format;
```

Reply:
+1 for renaming the `services` field. 
However, adding `topic` postfix especially to the `exclude_service_event_topics` might lead to a long name and sort of redundant since from the name `service_event` already clear that this is not the regular `service` name. 
I let you to decide with naming IMO it would be clear enough without adding `_topic` postfix.

Reply:
> However, adding `topic` postfix especially to the `exclude_service_event_topics` might lead to a long name and sort of redundant since from the name `service_event` already clear that this is not the regular `service` name. I let you to decide with naming IMO it would be clear enough without adding `_topic` postfix.

Okay. Let's keep it as it is!

> https://github.com/Barry-Xu-2018/rosbag2/blob/5c2006503d91ce47b29cf8b89782954fdf601d6b/rosbag2_py/src/rosbag2_py/_transport.cpp#L361
>
> Not use the same variable name.

I will change this for using the same name.

Reply:
@Barry-Xu-2018 Yes, please don't forget to change name here https://github.com/Barry-Xu-2018/rosbag2/blob/5c2006503d91ce47b29cf8b89782954fdf601d6b/rosbag2_py/src/rosbag2_py/_transport.cpp#L361 as well.

Reply:
@MichaelOrlov 

> please don't forget to change name here https://github.com/Barry-Xu-2018/rosbag2/blob/5c2006503d91ce47b29cf8b89782954fdf601d6b/rosbag2_py/src/rosbag2_py/_transport.cpp#L361

Yes. I submitted a commit  https://github.com/ros2/rosbag2/pull/1480/commits/097384066e5a5e626c683e7f8070fd98c2b887dc.

Reply:
Thanks! I missed it.